templates/service.mustache

```diff
@@ -22,31 +22,49 @@ export interface {{FUNCTION_NAME}}Params {
 	/**
 	 * {{FUNCTION_PARAM_DESCRIPTION}}
 	 */
-	{{FUNCTION_PARAM}}: string;
+	{{FUNCTION_PARAM}}{{^FUNCTION_PARAM_OPTIONAL}}?{{/FUNCTION_PARAM_OPTIONAL}}: string;
 	{{/FUNCTION_PARAM_CONFIGS}}
 }
 {{/FUNCTION_PARAMS}}
 
+{{#QUERY_PARAMS}}
+/**
+ * The query parameters for the {@link {{FUNCTION_NAME}}} function.
+ */
+export interface {{FUNCTION_NAME}}QueryParams {
+	{{#QUERY_PARAM_CONFIGS}}
+	/**
+	 * {{QUERY_PARAM_DESCRIPTION}}
+	 */
+	{{QUERY_PARAM}}{{^QUERY_PARAM_OPTIONAL}}?{{/QUERY_PARAM_OPTIONAL}}: string;
+	{{/QUERY_PARAM_CONFIGS}}
+}
+{{/QUERY_PARAMS}}
+
 /**
  * {{FUNCTION_SUMMARY}}
  * @param params - The request parameters (or undefined).
  * @param payload - the request payload (or undefined).
  * @param headers - the request headers.
  * @param cancelToken - optional cancel token to cancel this request.
  * @param basePathOverride - optional override to the base path.
+ * @param queryParameters - The request query parameters (or undefined).
```

Comment:
- [ ] Because this parameter is optional, we should mimic the function docs for the other optional params.

Reply:
- [ ] This argument should be before the basePathOverride

Reply:
Fixed